pyvrp/cpp/ProblemData.cpp

```diff
@@ -42,19 +62,52 @@ ProblemData::Client::Client(Coordinate x,
         throw std::invalid_argument("prize must be >= 0.");
 }
 
+ProblemData::Client::Client(Client const &client)
+    : x(client.x),
+      y(client.y),
+      demand(client.demand),
+      serviceDuration(client.serviceDuration),
+      twEarly(client.twEarly),
+      twLate(client.twLate),
+      releaseTime(client.releaseTime),
+      prize(client.prize),
+      name(duplicate(client.name)),
+      required(client.required)
+{
+}
+
+ProblemData::Client::Client(Client &&client)
+    : x(client.x),
+      y(client.y),
+      demand(client.demand),
+      serviceDuration(client.serviceDuration),
+      twEarly(client.twEarly),
+      twLate(client.twLate),
+      releaseTime(client.releaseTime),
+      prize(client.prize),
+      name(client.name),  // we can steal
+      required(client.required)
+{
+    client.name = nullptr;  // stolen
+}
+
+ProblemData::Client::~Client() { delete[] name; }
```

Comment:
Because we have to manage these raw C strings ourselves, I need to set up a lot more: explicit memory allocation, copy/move constructors, and the destructor. I'm not setting up the copy or move assignment operators, because these objects are essentially constant after construction.

All this works, but it's a bit hard to test from Python because I do not know exactly where copy/move get used. It's pretty easy to reason about correctness, however, even without explicit tests for these constructors.

[same story applies for VehicleType]